omnibus/package-scripts/angrychef/preinst

```diff
@@ -10,14 +10,5 @@
 INSTALLER_DIR=/opt/chef
 if [ -e $INSTALLER_DIR ]; then
   echo "removing $INSTALLER_DIR..."
-
-  # have to do this dance of moving /opt/chef to a tmp dir since files may be in use
-  tmp_dir="/opt/.chef.$$"
-  # if we can't create the tmp_dir then fail hard to prevent any possible security hole
-  (umask 077 && mkdir $tmp_dir) || exit 1
-  # now we can clean up the tmp_dir we created safely
-  mv $INSTALLER_DIR $tmp_dir
-  # ignore errors which must be EBUSY issues, this may crate some litter, which may
-  # be unavoidable
-  rm -rf $tmp_dir || true
+  rm -rf $INSTALLER_DIR || exit 1
```

Comment:
The problem here is there are some rare cases where this hits EBUSY (see https://github.com/chef/chef/pull/8046).

The original customer requesting this change hit an issue where all this data in `/opt` violated their LVM rules. Since there's probably only a small handful of files that would even hit EBUSY, we can do both:

```shell
if [ -e $INSTALLER_DIR ]; then
  echo "removing $INSTALLER_DIR..."
  rm -rf $INSTALLER_DIR || exit 1
fi

if [ -e $INSTALLER_DIR ]; then
  echo "Some files failed to delete, moving them so we don't interfere with upgrades, and then removing them"
  # have to do this dance of moving /opt/chef to a tmp dir since files may be in use
  tmp_dir="/opt/.chef.$$"
  # if we can't create the tmp_dir then fail hard to prevent any possible security hole
  (umask 077 && mkdir $tmp_dir) || exit 1
  # now we can clean up the tmp_dir we created safely
  mv $INSTALLER_DIR $tmp_dir
  # ignore errors which must be EBUSY issues, this may crate some litter, which may
  # be unavoidable
  rm -rf $tmp_dir || true
fi
```